syncer/syncer.go

```diff
@@ -974,6 +1015,10 @@ func (s *Syncer) sync(tctx *tcontext.Context, queueBucket string, db *DBConn, jo
 			if !ok {
 				return
 			}
+			if sqlJob.tp == xid {
+				lastAddedXidPos.save(sqlJob.location.Clone(), nil)
```

Comment:
Is `currentLocation` more accurate? (although `location` and `currentLocation` are the same for XID job).

Reply:
addressed in https://github.com/pingcap/dm/pull/605/commits/c1bef9fa9626eac62ce0509b7a558c2fbc379f97